internal/cache/file/cache_handle_test.go

```diff
@@ -636,8 +650,33 @@ func (cht *cacheHandleTest) Test_Read_FileInfoGenerationChanged() {
 	fileInfoData.ObjectGeneration = 1
 	err = cht.cache.UpdateWithoutChangingOrder(fileInfoKeyName, fileInfoData)
 	AssertEq(nil, err)
-	_, err = cht.cacheHandle.Read(context.Background(), cht.bucket, cht.object, 0, dst)
+	_, cacheHit, err = cht.cacheHandle.Read(context.Background(), cht.bucket, cht.object, 0, dst)
 
 	expectedErr := fmt.Errorf("%v: generation of cached object: %v is different from required generation: ", util.InvalidFileInfoCacheErrMsg, fileInfoData.ObjectGeneration)
 	AssertTrue(strings.Contains(err.Error(), expectedErr.Error()))
+	AssertFalse(cacheHit)
+}
+
+func (cht *cacheHandleTest) Test_MultipleReads_CacheHitShouldBeFalseThenTrue() {
+	dst := make([]byte, ReadContentSize)
+	offset := int64(cht.object.Size - ReadContentSize)
+	cht.cacheHandle.isSequential = true
+	cht.cacheHandle.prevOffset = offset - util.MiB
+	cht.cacheHandle.downloadFileForRandomRead = true
+
+	// First read should be cache miss.
+	n, cacheHit, err := cht.cacheHandle.Read(context.Background(), cht.bucket, cht.object, offset, dst)
+
+	jobStatus := cht.cacheHandle.fileDownloadJob.GetStatus()
+	ExpectGe(jobStatus.Offset, offset)
+	ExpectEq(n, ReadContentSize)
+	cht.verifyContentRead(offset, dst[:n])
+	ExpectFalse(cacheHit)
+	ExpectEq(nil, err)
+
```

Comment:
nit: to maintain consistency with other tests, let's give only one linespace. you can give after calling the second read and before the last three expect statements.